### ISSUE TITLE
Update the API to support proper classes, remove raw maps, class responses

### ISSUE THREAD

**Issue by bjornharvold** [cloudinary/cloudinary_java#239]
## Feature request for Cloudinary Java SDK
I think it's safe to say that most Java users today use >= JDK 8. I would be great to have a Cloudinary API you could use and know exactly what you need to send and exactly what you are going to get back without having to read the docs on what JSON responses return or what you could pass for the "Object" parameter in a method like upload().

Adding support for uploading InputStream and byte[] would also be a great addition.

## Explain your use case
Asking for a self-explanatory API

## Describe the problem you’re trying to solve
Not spending so much time going through the documentation on what raw Maps include. 

## Do you have a proposed solution?
- Instead of passing Objects, create convenience methods that take self-explanatory classes.
- Instead of raw maps, pass class objects with the proper / available accessors.

In a nutshell, make this API more friendly. Currently, every time I have to work with the Cloudinary Java API, I have to start from scratch and read the docs over again to see what's available to me. Look at Stripe's Java API for an example of an incredibly well-formed API.

**Comment by michalkcloudinay** [comment]
Thanks @bjornharvold for your suggestion. We will open a request internally to support this and will update here with any new insights.

**Comment by aksjoshi1** [comment]
After careful review, our team has decided not to go ahead with this request as it requires a breaking change.

**Comment by campbellbartlett** [comment]
+1 to this issue. I get that it might introduce a breaking change, but relying on Maps of Objects instead of proper POJOs poses a significant risk. If something gets renamed, there’s no way to catch it with compile-time checks when updating to a new release of the SDK.
